### PR TITLE
Added options to opt-out of AND/OR in filtering.

### PR DESCRIPTION
**Pull request by PascalSenn**
This PR adds a options to make AND and OR configureable
Before this PR there was no way to disable this feature globally. 
One would have to create all customer FilterInputTypes or create a type interceptor to remove the fields from the types.

# BREAKING CHANGE
- This PR adds two new members to the `IFilterConvention` interfaces. This would break exisiting implementation of this interface. As filtering is normally extended by extending FilterConvention, i do not think this interface was ever used directly.

### CODE REVIEW

File: src/HotChocolate/Data/src/Data/Filters/Convention/IFilterConventionDescriptor.cs [hunk 1]
```diff
@@ -111,5 +111,22 @@ IFilterConventionDescriptor AddProviderExtension<TExtension>()
...
+        IFilterConventionDescriptor AllowOr(bool allow = true);
+
+        /// <summary>
+        /// Defines if AND-combinators are allowed for filtering.
+        /// </summary>
+        /// <param name="allow">
+        /// Specifies if AND-combinators are allowed or disallowed.
+        /// </param>
+        IFilterConventionDescriptor AllowAnd(bool allow = true);
+
```
Comment (PascalSenn) [inline src/HotChocolate/Data/src/Data/Filters/Convention/IFilterConventionDescriptor.cs:130]: line

**Comment by michaelstaib** [comment]
Is this one ready?

**Comment by PascalSenn** [comment]
@michaelstaib  yes but it is breaking? it breaks a interface.. so, v12?

**Comment by michaelstaib** [comment]
Ah ... I did forget this. OK, lets keep it for now and see in the next meetings when we start work in V12.

File: src/HotChocolate/Data/src/Data/Filters/Convention/IFilterConventionDescriptor.cs (hunk 1 above)
Comment (michaelstaib) [inline src/HotChocolate/Data/src/Data/Filters/Convention/IFilterConventionDescriptor.cs:130]:
```suggestion
```

**Review by michaelstaib — APPROVED** [review]

File: src/HotChocolate/Data/test/Data.Filters.Tests/Convention/FilterConventionTests.cs [hunk 2]
```diff
@@ -387,6 +387,56 @@ public void FilterProvider_Throws_Exception_When_NotInitializedByConvention()
             exception.Message.MatchSnapshot();
         }
 
+        [Fact]
+        public async Task FilterConvention_Should_NotAdd()
```
Comment (arif-hanif) [inline src/HotChocolate/Data/test/Data.Filters.Tests/Convention/FilterConventionTests.cs:391]: FilterConvention_Should_NotAddAnd()

**Comment by PascalSenn** [comment]
@michaelstaib  i think this one can go into 12

File: src/HotChocolate/Data/test/Data.Filters.Tests/Convention/FilterConventionTests.cs (hunk 2 above)
Comment (michaelstaib) [reply src/HotChocolate/Data/test/Data.Filters.Tests/Convention/FilterConventionTests.cs:391]:
```suggestion
        public async Task FilterConvention_Should_NotAddAnd()
```

**Review by michaelstaib — APPROVED** [review]

**Pull request merged**